### PR TITLE
Migrated CommunitiesController to RQ

### PR DESCRIPTION
**Pull request by mzparacha**
## Link to Issue
Closes: https://github.com/hicommonwealth/commonwealth/issues/4975

## Description of Changes
Migrated CommunitiesController to RQ, it only had 1 mutation with help app-level state.

## "How We Fixed It"
By using react query for CommunitiesController mutations

## Test Plan
- Login and join some communities
- Open the explore menu, you will see a list of communities you have joined
- Click on the star icon next to a community to `star` it. Verify it gets starred i.e the star icon fills with yellow color
- Click on the same star icon again. Verify the community gets unstarred i.e the star icon only has a yellow border.

## Deployment Plan
N/A

## Other Considerations
Currently, the star/unstar community is happening by the `/starCommunity` API. I think its a good time to migrate it to be restful. @jnaviask for thoughts.

### CODE REVIEW

**Comment by mzparacha** [comment]
> Other Considerations
Currently, the star/unstar community is happening by the /starCommunity API. I think its a good time to migrate it to be restful. @jnaviask for thoughts.

@jnaviask Should I create a followup ticket and start on it as well?

File: packages/commonwealth/client/scripts/controllers/server/user.ts [hunk 1]
```diff
@@ -289,13 +289,21 @@ export class UserController {
...
   }
 
+  public isCommunityStarred(chain: string): boolean {
+    return (
+      this._starredCommunities.findIndex((c) => {
+        return c.chain === chain;
+      }) !== -1
+    );
+  }
+
```
Comment (mzparacha) [inline packages/commonwealth/client/scripts/controllers/server/user.ts:299]: Starred communities are calculated during app init. So for now added this method as it seemed the best option.

File: packages/commonwealth/client/scripts/state/api/communities/toggleCommunityStar.ts [hunk 2]
```diff
@@ -0,0 +1,51 @@
+import { useMutation } from '@tanstack/react-query';
+import axios from 'axios';
+import StarredCommunity from 'models/StarredCommunity';
+import app from 'state';
+
+interface ToggleCommunityStarProps {
+  chain: string;
+  isAlreadyStarred: boolean; // TODO: rename to `shouldStar`
```
Comment (mzparacha) [inline packages/commonwealth/client/scripts/state/api/communities/toggleCommunityStar.ts:8]: when we migrate api to be restful then we can rename prop

**Comment by jnaviask** [comment]
> > Other Considerations
> > Currently, the star/unstar community is happening by the /starCommunity API. I think its a good time to migrate it to be restful. @jnaviask for thoughts.
> 
> @jnaviask Should I create a followup ticket and start on it as well?

@mzparacha Yes, please make a follow-up ticket to make community-related routes RESTful.

**Comment by mzparacha** [comment]
> @mzparacha Yes, please make a follow-up ticket to make community-related routes RESTful.

Ticket [here](https://github.com/hicommonwealth/commonwealth/issues/4996). I also left a comment in the ticket for you [here](https://github.com/hicommonwealth/commonwealth/issues/4996#issuecomment-1710331360).

**Review by jnaviask — APPROVED** [review]
LGTM, let's get this into v0.6.9 unless @masvelio has other comments that block its release.

**Review by masvelio — APPROVED** [review]

**Pull request merged**